src/test/ui/macros/rfc-3086-metavar-expr/syntax-errors.stderr

```diff
@@ -196,6 +196,18 @@ error: unrecognized meta-variable expression
 LL |     ( $( $i:ident ),* ) => { ${ aaaaaaaaaaaaaa(i) } };
    |                                 ^^^^^^^^^^^^^^ help: supported expressions are count, ignore, index and length
 
+error: unexpected token: crate
+  --> $DIR/syntax-errors.rs:149:19
+   |
+LL |                 $$crate
+   |                   ^^^^^
+
+note: $$crate is not allowed
+  --> $DIR/syntax-errors.rs:149:19
+   |
+LL |                 $$crate
+   |                   ^^^^^
```

Comment:
The perfect span would be both `$ $` and `crate`. Scanning this, I think `tree` is the leading `$` and you get its span on L150 so you could make a (multispan? union? don't know the proper name) of that, the `$` here, and the lookahead `crate`.

It's worth also checking the diagnostics when further nested and indirected, since this is macro-expanded-macro territory.

Reply:
Changed to:

```
error: unexpected token: crate
  --> $DIR/dollar_dollar_crate.rs:9:19
   |
LL |                 $$crate
   |                   ^^^^^

note: `$$crate` is not allowed in any context
```

Not great but isn't wrong.